### PR TITLE
[CES-109] Switch Messages Cosmos secondary location to ItalyNorth

### PR DESCRIPTION
**Pull request by mamu0**
### List of changes

Replace NorthEurope with ItalyNorth region for data redundancy in read-only mode for messages namespace
Added Private endpoint for italy north.

### Motivation and context

ItalyNorth migration. This is a first step to have data in ItalyNorth without affecting operations on the main region, WestEurope

### Type of changes

- [ ] Add new resources
- [X] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [X] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: src/domains/citizen-auth-common/01_network.tf [hunk 1]
```diff
@@ -57,3 +62,18 @@ module "redis_common_snet" {
...
+
+## Cosmos Private Endpoint
+resource "azurerm_private_endpoint" "cosmos_db" {
+  name                = coalesce("${local.product}-citizen-auth-account", format("%s-private-endpoint-sql-itn", module.cosmosdb_account.name))
+  location            = "italynorth"
+  resource_group_name = azurerm_resource_group.data_rg.name
+  subnet_id           = data.azurerm_subnet.private_endpoints_subnet_itn.id
+
+  private_service_connection {
+    name                           = coalesce("${local.product}-citizen-auth-account-private-endpoint", format("%s-private-endpoint-sql", module.cosmosdb_account.name))
```
Comment (Krusty93) [inline src/domains/citizen-auth-common/01_network.tf:74]: Please use the new naming convention removing old values

File: src/domains/citizen-auth-common/.terraform.lock.hcl [hunk 2]
```diff
@@ -2,58 +2,49 @@
 # Manual edits may be lost in future updates.
 
 provider "registry.terraform.io/hashicorp/azuread" {
-  version     = "2.33.0"
-  constraints = "<= 2.33.0"
+  version     = "2.53.0"
```
Comment (Krusty93) [inline src/domains/citizen-auth-common/.terraform.lock.hcl:5]: I think some architecture is missing, run `terraform providers lock -platform=windows_amd64 -platform=darwin_amd64 -platform=darwin_arm64 -platform=linux_amd64` to check for it

File: src/domains/citizen-auth-common/01_network.tf [hunk 3]
```diff
@@ -46,14 +46,34 @@ data "azurerm_private_dns_zone" "privatelink_table_core" {
...
   address_prefixes     = var.cidr_subnet_redis_common
   resource_group_name  = local.vnet_common_resource_group_name
   virtual_network_name = local.vnet_common_name
 
   private_endpoint_network_policies_enabled = false
 }
+
+## Cosmos Private Endpoint
+resource "azurerm_private_endpoint" "cosmos_db" {
+  name                = "${local.product}-itn-citizen-auth-account"
```
Comment (Krusty93) [inline src/domains/citizen-auth-common/01_network.tf:68]: I would prefer to stick to naming convention, something like:
` ${project}${region}${domain}${app_name}-sql-pep-${instance_number}`

File: src/domains/citizen-auth-common/01_network.tf [hunk 4]
```diff
@@ -46,14 +46,34 @@ data "azurerm_private_dns_zone" "privatelink_table_core" {
...
+  location            = "italynorth"
+  resource_group_name = azurerm_resource_group.data_rg.name
+  subnet_id           = data.azurerm_subnet.private_endpoints_subnet_itn.id
+
+  private_service_connection {
+    name                           = "${local.product}-itn-citizen-auth-account-private-endpoint"
```
Comment (Krusty93) [inline src/domains/citizen-auth-common/01_network.tf:74]: Ditto, same value as above

**Review by Krusty93 — CHANGES_REQUESTED** [review]
`module.lollipop_assertions_storage.azurerm_advanced_threat_protection.this` is going to be deleted

EDIT: it does make sense if these variables have the following values:
`advanced_threat_protection: true`
`use_legacy_defender_version: false`

File: src/domains/citizen-auth-common/01_network.tf (hunk 4 above)
Comment (BurnedMarshal) [reply src/domains/citizen-auth-common/01_network.tf]: fixed in [7c6c596](https://github.com/pagopa/io-infra/pull/1188/commits/7c6c5968af2b018054b3d6cb89f3cc73ad568177)

File: src/domains/citizen-auth-common/01_network.tf (hunk 3 above)
Comment (BurnedMarshal) [reply src/domains/citizen-auth-common/01_network.tf]: fixed in [7c6c596](https://github.com/pagopa/io-infra/pull/1188/commits/7c6c5968af2b018054b3d6cb89f3cc73ad568177)

**Review by BurnedMarshal — APPROVED** [review]
LGTM

**Review by Krusty93 — APPROVED** [review]

**Pull request merged**